lib/ansible/modules/cloud/amazon/aws_organization_units.py

```diff
@@ -0,0 +1,270 @@
+    type: str
+  arn:
+    description:
+      - ARN of the organizational unit
+      - ARN can be used to delete OU but not create
+      - Mutually exclusive with c(name)
+    required: true
+    type: str
+  state:
+    description:
+      - State of the organizational unit
+    default: present
+    type: str
+    choices:
+      - present
+      - absent
+extends_documentation_fragment:
+  - aws
+  - ec2
+requirements:
+  - boto3
+  - botocore
+"""
+
+EXAMPLES = """
+- name: Create organizational unit
+  aws_organizations:
+    name: Prod
+    state: present
+
+- name: Delete organizational unit
+  aws_organizations:
+    name: Prod
+    state: absent
+"""
+
+RETURN = """
+name:
+  description:
+    - Name of the organizational unit.
+      If the parent OU does not exist the module will fail with warning message.
+  returned: always
+  type: str
+  sample: Prod
+arn:
+  description:
+    - ARN of the organizational unit.
+      If the parent OU does not exist the module will fail with warning message.
+  returned: always
+  type: str
+  sample: Prod
+state:
+  description: State of the organization unit.
+  returned: always
+  type: str
+  sample: present
+ou:
+  description: The Organization Unit details.
+  returned: always
+  type: dict
+  sample:
+    {
+        "Arn": "arn:aws:organizations::account:ou/o-id/ou-id",
+        "Id": "ou-id",
+        "Name": "Prod"
+    }
+"""
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    # handled by AnsibleAWSModule
+    pass
+
+from ansible.module_utils.ec2 import AWSRetry, boto3_tag_list_to_ansible_dict, ansible_dict_to_boto3_tag_list, camel_dict_to_snake_dict
+from ansible.module_utils.aws.core import AnsibleAWSModule, is_boto3_error_code
+from ansible.module_utils._text import to_native
+
+class ParameterValidationException(Exception):
+    pass
+
+class UnknownConditionException(Exception):
+    pass
+
+class OrganizationalUnitDependencyException(Exception):
+    pass
+
+class AwsOrganizationalUnit():
+    def __init__(self, name=None, arn=None):
+        self.arn = arn
+        self.client = boto3.client("organizations")
+        self.root = self.get_aws_organization_root()
+        if arn:
+            self.name = name
+            self.ou = self.get_aws_organizational_unit_by_arn(self.arn)
+        elif name:
+            self.name = name.strip("/")
+            self.ou = self.get_aws_organizational_unit_by_name(self.name)
+        else:
+            raise ParameterValidationException
+
+    def get_aws_organization_root(self):
+        roots = self.client.list_roots()
+        if len(roots.get("Roots")) != 1:
+            raise UnknownConditionException
+        return roots.get("Roots")[0]
+
+    def get_aws_organizational_unit_for_parent(self, ou_name, parent_id):
+        paginator = self.client.get_paginator("list_organizational_units_for_parent")
+        for page in paginator.paginate(ParentId=parent_id):
```

Comment:
```suggestion
        parent_ous = paginator.paginate(ParentId=parent_id).build_full_result().get('OrganizationalUnits', [])
        for child_ou in parent_ous:
```
You could remove the outer loop here by using the paginator's `.build_full_result()` method and retrieving the OrganizationalUnits from it.

Reply:
That looks better thanks